File: php/mortgage-calc-restclient.php
<?php

// constants
const PATH_TO_CLASSES = ".class";

// user classes autoloading
spl_autoload_register(function($class_name)
{
  $class_source_file_path = PATH_TO_CLASSES."/".$class_name.".php";
  require_once $class_source_file_path;
});

const MESSAGE_ERROR_CONTACTING_SERVICE = "error contacting the REST service";
const MESSAGE_ERROR_DECODING_JSON = "error decoding the reply JSON";
const MESSAGE_RECEIVED_ERROR = "received error";

$loan_amount = 350000.0;
$loan_term_months = 12;
$interest_rate_percent_pa = 9.00;

$requestJson = tJson::codeCalculation($loan_amount, $loan_term_months, $interest_rate_percent_pa);

$replyJson = tWeb::request($requestJson);
if ($replyJson === null)
{
  echo " - ".MESSAGE_ERROR_CONTACTING_SERVICE.PHP_EOL;
  return;
}

$replyData = tJson::decodeResultCalculation($replyJson);
if ($replyData === null)
{
  $errorString = tJson::getLastErrorString();

  if ($errorString != null)
  {
    echo " - ".MESSAGE_RECEIVED_ERROR.": ".$errorString.PHP_EOL;
  }
  else
  {
    echo " - ".MESSAGE_ERROR_DECODING_JSON.PHP_EOL;
  }

  return;
}

echo " - interest rate % p.m.: ".$replyData[tJson::INTEREST_RATE_PERCENT_PM].PHP_EOL;
echo " - discont factor: ".$replyData[tJson::DISCONT_FACTOR].PHP_EOL;
echo " - monthly payment: ".$replyData[tJson::MONTHLY_PAYMENT].PHP_EOL;
echo " - total paid: ".$replyData[tJson::TOTAL_PAID].PHP_EOL;

foreach ($replyData[tJson::SCENARIO] as $index => $oneMonth)
{
  echo PHP_EOL;

  echo " - month number: ".($index + 1).PHP_EOL;
  echo "   - payment: ".$oneMonth[tJson::PAYMENT].PHP_EOL;
  echo "   - interest: ".$oneMonth[tJson::INTEREST].PHP_EOL;
  echo "   - amortization: ".$oneMonth[tJson::AMORTIZATION].PHP_EOL;
  echo "   - account balance: ".$oneMonth[tJson::ACCOUNT_BALANCE].PHP_EOL;
}

?>

File: python/MortgageCalcRestClient.py
from classes.Web import Web
from classes.Json import Json

MESSAGE_ERROR_CONTACTING_SERVICE = "error contacting the REST service"
MESSAGE_ERROR_DECODING_JSON = "error decoding the reply JSON"
MESSAGE_RECEIVED_ERROR = "received error"

loan_amount = 350000.0
loan_term_months = 12
interest_rate_percent_pa = 9.00

requestJsonCalculation = Json.codeCalculation(loan_amount, loan_term_months, interest_rate_percent_pa)

try:
    replyJsonCalculation = Web.request(requestJsonCalculation)
except:
    print(" - " + MESSAGE_ERROR_CONTACTING_SERVICE)
    exit()

replyData = Json.decodeResult(replyJsonCalculation)
if replyData == None:
    errorString = Json.getLastErrorString()

    if errorString != None:
        print(" - " + MESSAGE_RECEIVED_ERROR + ": " + errorString)
    else:
        print(" - " + MESSAGE_ERROR_DECODING_JSON)

    exit()

print(" - interest rate % p.m.:", replyData[Json.INTEREST_RATE_PERCENT_PM])
print(" - discont factor:", replyData[Json.DISCONT_FACTOR])
print(" - monthly payment:", replyData[Json.MONTHLY_PAYMENT])
print(" - total paid:", replyData[Json.TOTAL_PAID])

index = 1
for oneMonth in replyData[Json.SCENARIO]:
    print()

    print(" - month number:", index)
    print("   - payment:", oneMonth[Json.PAYMENT])
    print("   - interest:", oneMonth[Json.INTEREST])
    print("   - amortization:", oneMonth[Json.AMORTIZATION])
    print("   - account balance:", oneMonth[Json.ACCOUNT_BALANCE])

    index = index + 1
